basics/inspecting-the-vivaldi-ui-with-devtools.md

```diff
@@ -0,0 +1,95 @@
+
+1. Open devtools and click the element picker button (mouse pointer) on top left. It turns blue.
+2. Hover the active tab in the Vivaldi window and move around until you see the span element with the `.title` class.
+
+    ![inspect.png](assets/inspect.png) 
+
+3. Click to select. The element is being highlighted in the “Elements” tab, which represents the page source of the user interface.
+4. A second section inside the tab contains the “Styles” tab either at the side or beneath. In the source we can find the selectors, the style shows the CSS acting on the selected element. Scroll down until you find the command that changes the color of the active tab.
+
+    ![edit-1.png](assets/edit-1.png) 
+
+5. Click on `var(--colorAccentFg)` (this is an example, the color could be different in your theme), press `backspace`, input `red` and press `enter`. This changes the font color, but only temporary (until reload).
+6. Should you have the experiment for using CSS modifications enabled in `vivaldi://experiments/`, you can both access and edit your modification files in the developer tools. Click on the “Sources” tab and in the sidebar expand `vivaldi-data` and then `css-mods`. Click on your mod file, the contents will be shown right next to it. Since we know how Vivaldi changes the color of the active tab, we can use the same code in our file to overwrite Vivaldi’s instructions. Change the color to `green` this time and notice how the change is immediately visible.
+
+    ![edit-2.png](assets/edit-2.png) 
+
+7. This is a temporary modification too. However, we can make our edit permanent by right‐clicking and selecting “Save as...”. Choose the location of the mod file to overwrite, or save the edit as backup.
+
+---
+
+### Reloading the user interface
+
+The standard way of performing edits is opening our mod file in a proper text editor, editing, saving and reloading the UI. Here a few options to load the changes:
+
+- Exit Vivaldi and open it again.
+- Open a new browser window.
+- Select all tabs and move them to a new window. This is available as a menu entry when right‐clicking a tab and has the advantage of keeping sites and media playback active.
+- Restart Vivaldi. The URL **`vivaldi://restart/`** will restart Vivaldi when it is being opened in the current tab. We can either enter the address directly in the address field, or use a bookmark. One nifty option is to save the address, assign a nickname to it (e.g. `re`) and load it with quick commands. `shift‐/alt-enter` will open in current tab, when bookmarks are set to open in a new tab by default and vice versa; this is dependent on our search, bookmarks and quick commands settings.
+
+All the actions load the `browser.html` file anew, which in turn loads our modification files.
+
+---
+
+### Console
+
+Let’s write a “Hello, World!” example script for testing our Javascript setup as described in [Modding Vivaldi](https://forum.vivaldi.net/topic/10549/modding-vivaldi).
```

Comment:
1. Maybe mention that you can skip this part if you do not plan to make JS mods?
2. Maybe link to ../using-mods/installation.md instead of the forum topic?

Reply:
1.  no, you make no sense
2.  yes